internal/bminventory/inventory.go

```diff
@@ -2535,6 +2535,7 @@ func (b *bareMetalInventory) updateNetworkTables(db *gorm.DB, cluster *common.Cl
 func (b *bareMetalInventory) updatePlatformParams(params installer.V2UpdateClusterParams, updates map[string]interface{}, usages map[string]models.Usage) error {
 	if params.ClusterUpdateParams.Platform != nil && common.PlatformTypeValue(params.ClusterUpdateParams.Platform.Type) != "" {
 		updates["platform_type"] = params.ClusterUpdateParams.Platform.Type
+		updates["platform_is_external"] = swag.BoolValue(params.ClusterUpdateParams.Platform.IsExternal)
```

Comment:
I see, we update the DB directly from the user params struct instead of constructing a cluster struct in between to apply the business rules, your suggestion makes more sense in this way.

Reply:
As discussed, it looks like the best way to update the flag is in `GetActualUpdateClusterPlatformParams`/`GetActualCreateClusterPlatformParams` as these functions contain the business logic to return a valid `Platform` struct.